helm/prometheus-operator-app/templates/_helpers.tpl

```diff
@@ -165,7 +165,21 @@ Use the prometheus-node-exporter namespace override for multi-namespace deployme
   {{- end -}}
 {{- end -}}
 
+{{- define "kube-prometheus-stack.crdInstall" -}}
+{{- printf "%s-%s" ( include "kube-prometheus-stack.name" . ) "crd-install" | replace "+" "_" | trimSuffix "-" -}}
+{{- end -}}
+
+{{- define "kube-prometheus-stack.CRDInstallAnnotations" -}}
+"helm.sh/hook": "pre-upgrade"
```

Comment:
Only run on upgrade. For new install, `prometheus-operator-crd` need to be installed first.